### PR TITLE
Refactor admin notifications and settings.

### PR DESCRIPTION
**Pull request by rdiepenbrock**
Refactor and clean up:

1. Combine admin notifications
2. Add close button to notifications
3. Add multisite check for settings page
4. Enable settings page and test configuration for all WP install types

@pirogoeth take a good look at this one, if you can. With the changes made, I'd like to be sure nothing got missed.
I've tested using both the wp-config constants and the settings page.

### CODE REVIEW

File: includes/admin.php [hunk 1]
```diff
@@ -65,10 +63,12 @@ public function __construct()
...
         $sitename = strtolower($_SERVER['SERVER_NAME']);
-        if (substr($sitename, 0, 4) == 'www.') {
+        if (substr($sitename, 0, 4) == 'www.'):
             $sitename = substr($sitename, 4);
-        }
-        $regionDefault = (defined('MAILGUN_REGION') && MAILGUN_REGION) ? MAILGUN_REGION : $this->get_option('region');
+        endif;
+
+        $region = (defined('MAILGUN_REGION') && MAILGUN_REGION) ? MAILGUN_REGION : $this->get_option('region');
+		$regionDefault = $region ?: 'us';
```
Comment (pirogoeth) [inline includes/admin.php]: consistency between tabs and spaces?

File: includes/admin.php [hunk 2]
```diff
@@ -319,56 +311,51 @@ public function validation($options)
...
         $screen = get_current_screen();
-        if (!current_user_can('manage_options') || $screen->id == $this->hook_suffix
-        ) {
+        if (!current_user_can('manage_options') || $screen->id == $this->hook_suffix):
             return;
-        }
+        endif;
 
-        if ((!$this->get_option('apiKey') && $this->get_option('useAPI') === '1')
+        if (defined('WP_ALLOW_MULTISITE') && !defined('MAILGUN_REGION')
```
Comment (pirogoeth) [inline includes/admin.php]: please explicitly group these to make it more clear: 
```
if ( (defined('WP_ALLOW_MULTISITE') && !defined('MAILGUN_REGION'))
    || (!$this->get_option('apiKey') && $this->get_option('useAPI') === '1') ):
```

File: includes/admin.php [hunk 3]
```diff
@@ -319,56 +311,51 @@ public function validation($options)
...
-        }
-
-        if ($this->get_option('override-from') === '1'
-            && (!$this->get_option('from-name')
-            || !$this->get_option('from-address'))
-        ) {
-            ?>
-            <div id='mailgun-warning' class='notice notice-warning fade'>
+        if ($this->get_option('override-from') === '1' &&
+			(!$this->get_option('from-name') || !$this->get_option('from-address'))
```
Comment (pirogoeth) [inline includes/admin.php]: consistency between tabs and spaces here

File: includes/admin.php [hunk 4]
```diff
@@ -319,56 +311,51 @@ public function validation($options)
...
+?>
+			<div id='mailgun-warning' class='notice notice-warning is-dismissible'>
                 <p>
                     <strong>
                         <?php _e('Mailgun is almost ready. ', 'mailgun'); ?>
                     </strong>
-                    <?php printf(__('You must <a href="%1$s">configure Mailgun</a> for it to work.', 'mailgun'), menu_page_url('mailgun', false)); ?>
+	                <?php
+						printf(
+	                		__('With the latest update, Mailgun now supports multiple regions! By default, we will use the U.S. region, but we now have an EU region available. You can configure your Mailgun settings <a href="%1$s">here</a> or in your wp-config.php.', 'mailgun'),
```
Comment (pirogoeth) [inline includes/admin.php]: consistency between tabs and spaces here
tabs and spaces mixed as well

File: includes/options-page.php [hunk 5]
```diff
@@ -22,39 +22,47 @@
...
-            <span class="alignright"><a target="_blank" href="http://www.mailgun.com/"><img src="https://assets.mailgun.com/img/mailgun.svg" alt="Mailgun" style="width: 10em;"/></a></span>
+            <span class="alignright">
+				<a target="_blank" href="http://www.mailgun.com/">
+					<img src="https://assets.mailgun.com/img/mailgun.svg" alt="Mailgun" style="width:10em;"/>
+				</a>
+			</span>
             <h2><?php _e('Mailgun', 'mailgun'); ?></h2>
-            <p>A <a target="_blank" href="http://www.mailgun.com/">Mailgun</a> account is required to use this plugin and the Mailgun service.</p>
-            <p>If you need to register for an account, you can do so at <a target="_blank" href="http://www.mailgun.com/">http://www.mailgun.com/</a>.</p>
+            <p>
```
Comment (pirogoeth) [inline includes/options-page.php]: start and end tags not aligned?

File: includes/options-page.php [hunk 6]
```diff
@@ -194,6 +202,27 @@
...
                 </table>
+			<?php else: ?>
+				<p>
+					<?php
+						_e('Once you have configured the plugin settings in your wp-config.php, you can test it here.', 'mailgun');
+					?>
+				</p>
+				<p><?php _e('The definitions are as follows:', 'mailgun'); ?></p>
+				<p><pre>
+MAILGUN_REGION       Choices: 'us' or 'eu'
```
Comment (pirogoeth) [inline includes/options-page.php]: Maybe add the descriptions of these options from the readme?

File: includes/options-page.php [hunk 7]
```diff
@@ -214,15 +247,27 @@
                             <?php _e('Lists', 'mailgun'); ?>
                         </th>
                         <td>
-                            <a href="?page=mailgun-lists">View available lists</a>
+                            <?php
+                            	_e('<a href="?page=mailgun-lists">View available lists</a>', 'mailgun');
```
Comment (pirogoeth) [inline includes/options-page.php]: should the entire tag be in the translation function or just "View available lists"?

File: includes/options-page.php [hunk 8]
```diff
@@ -214,15 +247,27 @@
...
                     </tr>
                 </table>
 
-                <p><?php _e('Before attempting to test the configuration, please click "Save Changes".', 'mailgun'); ?></p>
+				<?php if(!$isMultisite): ?>
+                <p>
+					<?php
+						_e('Before attempting to test the configuration, please click "Save Changes".', 'mailgun');
+					?>
+				</p>
```
Comment (pirogoeth) [inline includes/options-page.php]: consistency between tabs and spaces here

File: mailgun.php [hunk 9]
```diff
@@ -326,18 +329,20 @@ public function list_form($list_address, $args = array(), $instance = array())
...
 
-                <?php if (count($list_addresses) > 1) : ?>
+                <?php if (count($list_addresses) > '1'): ?>
                     <ul class="mailgun-lists" style="list-style: none;">
-                        <?php foreach ($all_list_addresses as $la) : ?>
-                            <?php if (!in_array($la['address'], $list_addresses)) {
-            continue;
-        } ?>
+                        <?php
+							foreach ($all_list_addresses as $la):
```
Comment (pirogoeth) [inline mailgun.php]: consistency between tabs and spaces here

File: includes/options-page.php (hunk 5 above)
Comment (rdiepenbrock) [reply includes/options-page.php]: line 31 is the beginning of the first <p> (previously line 27).

File: includes/options-page.php (hunk 6 above)
Comment (rdiepenbrock) [reply includes/options-page.php]: I had originally copied them from the readme, then made some updates. I will update the text in the readme file to match.

**Pull request merged**